In [1]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import openpyxl

In [2]:
df=pd.read_excel("fortune500.xlsx")
df.head()

,Year,Rank,Company,Revenue (in millions),Profit (in millions)
0,1955,1,General Motors,9823.5,806
1,1955,2,Exxon Mobil,5661.4,584.8
2,1955,3,U.S. Steel,3250.4,195.4
3,1955,4,General Electric,2959.1,212.6
4,1955,5,Esmark,2510.8,19.1


# Q.EVALAUTE DESCRIPTIVE STATISTICS AND PROVIDE THE INFERENCES FROM CALCULATIONS W.R.T REVENUE

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29500 entries, 0 to 29499
Data columns (total 5 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Year                   29500 non-null  int64  
 1   Rank                   29500 non-null  int64  
 2   Company                29500 non-null  object 
 3   Revenue (in millions)  29500 non-null  float64
 4   Profit (in millions)   29500 non-null  object 
dtypes: float64(1), int64(2), object(2)
memory usage: 1.1+ MB


In [4]:
df.isnull().sum()

Year                     0
Rank                     0
Company                  0
Revenue (in millions)    0
Profit (in millions)     0
dtype: int64

1.No null values are present.

In [9]:
df1=df.copy()
df2=df1.groupby('Company')['Revenue (in millions)'].sum()
df2.sort_values(ascending=False)

Company
Exxon Mobil             5152459.6
General Motors          4750335.8
Ford Motor              3757198.6
Wal-Mart Stores         3366203.4
General Electric        2685105.6
                          ...    
Liquid Carbonic              51.4
Truax-Traer Coal             50.5
Bullard                      50.2
Cuban-American Sugar         50.1
Goodall-Sanford              49.7
Name: Revenue (in millions), Length: 2588, dtype: float64

Exxon Mobil total revenue is 5152459.6 millions which is greater than General Motors.

In [28]:
df1['Profit (in millions)']=pd.to_numeric(df1['Profit (in millions)'])

ValueError: Unable to parse string "N.A." at position 228

ValueError: Unable to parse string "N.A." at position 228 .
There is a typing error at position 228 of df1['Profit (in millions)'] which may be corrected.

In [32]:
df1['Profit (in millions)']=pd.to_numeric(df1['Profit (in millions)'],errors = 'coerce')

In [33]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29500 entries, 0 to 29499
Data columns (total 5 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Year                   29500 non-null  int64  
 1   Rank                   29500 non-null  int64  
 2   Company                29500 non-null  object 
 3   Revenue (in millions)  29500 non-null  float64
 4   Profit (in millions)   29131 non-null  float64
dtypes: float64(2), int64(2), object(1)
memory usage: 1.1+ MB


In [34]:
df3=df1.groupby('Company')['Profit (in millions)'].sum()
df3.sort_values(ascending=False)

Company
Exxon Mobil                     403684.1
General Electric                248127.3
Altria Group                    150823.8
Intl. Business Machines         150582.7
Bank of America Corp.           127315.0
                                  ...   
Freddie Mac                     -29909.0
Qwest Communications            -36415.5
JDS Uniphase                    -56121.9
Time Warner                     -92864.0
American International Group    -93089.0
Name: Profit (in millions), Length: 2588, dtype: float64

Exxon Mobil total profit is 403684.1 millions which is greater than General Motors (248127.3 millions ).

In [50]:
df1[df1['Year']== 2009]

,Year,Rank,Company,Revenue (in millions),Profit (in millions)
28500,2009,1,Exxon Mobil,442851.0,45220.0
28501,2009,2,Wal-Mart Stores,405607.0,13400.0
28502,2009,3,Chevron,263159.0,23931.0
28503,2009,4,ConocoPhillips,230764.0,-16998.0
28504,2009,5,General Electric,183207.0,17410.0
...,...,...,...,...,...
29495,2009,996,Tellabs,1729.0,-930.1
29496,2009,997,Administaff,1724.4,45.8
29497,2009,998,Sanderson Farms,1723.6,-43.1
29498,2009,999,MGIC Investment,1721.5,-518.9


Conclusions: 
1.No null values are present in given dataset.
2.There is a typing error at position 228 of df1['Profit (in millions)'] which was corrected.
3.Exxon Mobil total revenue is 5152459.6 millions which is greater than General Motors.
4.Exxon Mobil total profit is 403684.1 millions which is greater than General Motors (248127.3 millions ).
5.Exxon Mobil Revenue (in millions) is 442851.0 millions and profit 45220.0 which is millions which is ranked 1 in year 2009.
6.Overall Exxon Mobil's ranking as 1 is justifiable.